### ISSUE TITLE
[accessibility issue] HelpModal keyboard shortcut tables are difficult to understand for screen reader users

### ISSUE THREAD

**Issue by ckbedwell** [grafana/grafana#79516]
**Steps to reproduce**

Open up the help modal with either `h` ([an issue filed in itself](https://github.com/grafana/grafana/issues/74841)) or using `shift + ?` to open the help modal. It shows four html tables with keyboard shortcuts categorised inside them.

**The problems**

"Global", "Dashboard", "Focused Panel", and "Time Range" are marked up as the table headers (spanning two columns), however they do not serve as the table headers of the table (instead they serve as the caption of the table).

No tables have a programmatically associated label or accessible name:

The tables in the keyboard shortcuts dialog do not have accessible names. Instead the visible labels for the tables are marked up as the table headers for each table.

![Screenshot of Grafana with the Help dialog open with the VoiceOver caption box beside it. It shows that the first item in the table doesn't have a header associated with it.](https://github.com/grafana/grafana/assets/6906380/7f60f3d2-2587-4cd0-8bb8-147d005b384b)

UHG Audit IDs:
1.3112
1.3114

**Comment by amy-super** [comment]
@ckbedwell I was about to open a ticket to redesign this modal as a whole - because visually the misaligned groupings have alway bugged me. There are also some casing issues (`Remove Panel` vs `Toggle panel legend`) and I believe there are also undocumented shortcuts. 

I'm not exactly sure what to do with it, but want to align on how you plan to tackle this and #74841 - has a team picked these up or are they up for grabs?

**Comment by ckbedwell** [comment]
As far as I know nobody has picked it up --  I'm assuming this work would go to @yaelleC's team? I can certainly advise or do some review, just let me know and I'll help however I can 😃

**Comment by ckbedwell** [comment]
Going to piggy-back off this one to add that the text in light-mode doesn't meet contrast ratio requirements either, so take a look at that too in your redesign:

![A magnified screenshot of the text in the help modal which does not meet the contrast requirements.](https://github.com/grafana/grafana/assets/6906380/42b280f3-a76d-423a-9b0b-710bb9595500)


![A screenshot of Chrome devtools color picker. It shows a contrast ratio of 3.04 with the current color and background.](https://github.com/grafana/grafana/assets/6906380/756e8f58-6d04-4ff3-a084-42336ee55b7d)
